### DISCUSSION TITLE
Installation Ease and UUIDs

### DISCUSSION THREAD

**Discussion by jasonjgardner** [jasonjgardner/jg-rtx#17] [Feedback]
### Generate new UUID with each release?

**Pros**
- One-click install through .mcpack
- Keeps old texture designs available, if you're into that...

**Cons**
- Could lead to cluttered, leftover releases installed in Minecraft
- I might forget to do it

### Do not change UUID?

**Pros**
- Less work, less changes

**Cons**
- Resource pack must be uninstalled before installing newer releases/updates.
- Allows for mismatched JG RTX releases in multiplayer

---

Please share your thoughts!

**Comment by Shmew** [comment]
I don't have a preference either way, but I'd recommend doing it however marketplace packs do it (especially if you plan to have it there in the future). 

One thing I'd point out is that:

> I might forget to do it

You can automate the generation of the uuid as part of your build scripts.

**Reply by jasonjgardner** [reply]
Thanks for the feedback!

Sure, I _could_ automate it, but then I'd have to go write [some kind of automation script](https://github.com/jasonjgardner/jg-rtx/pull/19)...

Marketplace resource packs probably use the same UUID, buuut they also get updated through Minecraft. Updating a resource pack from outside the marketplace requires a manual installation, as far as I know. You'd get an error message if you attempt to install an .mcpack whose UUID already exists in Minecraft. So you'd have to uninstall and reinstall JG RTX every time there's a new release – which is probably the proper way to do it, but I suspect some users might find that additional step annoying.
